apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -223,26 +212,17 @@ class PageTemplateModal extends Component {
 	}
 
 	getTemplateGroups = () => {
-		const [ homepageTemplates, defaultTemplates ] = partition( this.props.templates, {
-			category: 'home',
-		} );
-
-		const currentThemeTemplate =
-			find( this.props.templates, { slug: this.props.theme } ) ||
-			find( this.props.templates, { slug: DEFAULT_HOMEPAGE_TEMPLATE } );
-
-		if ( ! this.props.isFrontPage || ! currentThemeTemplate ) {
-			return { homepageTemplates: sortBy( homepageTemplates, 'title' ), defaultTemplates };
-		}
-
-		const otherHomepageTemplates = reject( homepageTemplates, { slug: currentThemeTemplate.slug } );
-
-		const sortedHomepageTemplates = [
-			currentThemeTemplate,
-			...sortBy( otherHomepageTemplates, 'title' ),
-		];
-
-		return { homepageTemplates: sortedHomepageTemplates, defaultTemplates };
+		return {
+			blankTemplate: filter( this.props.templates, { slug: 'blank' } ),
+			aboutTemplates: filter( this.props.templates, { category: 'about' } ),
+			blogTemplates: filter( this.props.templates, { category: 'blog' } ),
+			contactTemplates: filter( this.props.templates, { category: 'contact' } ),
+			menuTemplates: filter( this.props.templates, { category: 'menu' } ),
+			portfolioTemplates: filter( this.props.templates, { category: 'portfolio' } ),
+			servicesTemplates: filter( this.props.templates, { category: 'services' } ),
+			teamTemplates: filter( this.props.templates, { category: 'team' } ),
+			homepageTemplates: sortBy( filter( this.props.templates, { category: 'home' } ), 'title' ),
+		};
 	};
 
 	renderTemplatesList = ( templatesList, legendLabel ) => (
```

Comment:
I think this method should be updated so that it returns null if `templatesList.length === 0`. Otherwise it renders a section even if there are no templates to display